anitacheung83/Timetable-Wallpaper
# Feature/pop up img

## Pull Request: Feature - Pop up images (Download Handling Enhancement)

### Description

This pull request introduces an enhancement to the existing `DownloadButton` and  new `ImgPopUp` components. The new feature allows users to download the timetable wallpaper images directly to their laptops when using a laptop/desktop device. For mobile devices and tablets, a modal is displayed with the generated timetable wallpaper image, enabling users to download it by long-pressing the image and selecting the "Save Image" option.

### Changes Made

- **`DownloadButton` Component:**
  - Added logic to differentiate between laptop/desktop and mobile/tablet devices.
  - Improved the `handleDownload` function to use the appropriate download function based on the device type.
  
- **`ImgPopUp` Component:**
  - Updated the `generateBase64Image` function to return a downloadable image for laptop/desktop users.
  - Added a modal for displaying the timetable wallpaper image on mobile/tablet devices.
  - Updated the image display and added download instructions for the modal.

### How to Test

1. Open the application on a laptop/desktop device.
2. Navigate to the page with the timetable.
3. Click the "Download" button.
4. Verify that the timetable wallpaper image is downloaded directly to your laptop/desktop.

5. Open the application on a mobile device or tablet.
6. Navigate to the page with the timetable.
7. Click the "Download" button.
8. Verify that a modal appears with the generated timetable wallpaper image.
9. Long-press the image inside the modal and select the "Download" option.
10. Verify that the image is downloaded to the device.

### Additional Notes

- Ensure that the application is responsive and handles various screen sizes appropriately.
- Confirm that the new features do not negatively impact existing functionality.

### Screenshots
<img width="712" alt="Screenshot 2023-11-15 at 2 03 00 PM" src="https://github.com/anitacheung83/Timetable-Wallpaper/assets/77123829/3f448643-a800-472d-8762-822f85d1c2cb">

### Checklist
- [x] Tests have been added or updated to cover the new functionality.
